EasyCommands.Tests/ScriptTests/RocketVolleyTests.cs

```diff
@@ -0,0 +1,145 @@
+﻿using Microsoft.VisualStudio.TestTools.UnitTesting;
+using System;
+using Moq;
+using Sandbox.ModAPI.Ingame;
+using VRageMath;
+using static EasyCommands.Tests.ScriptTests.MockEntityUtility;
+
+namespace EasyCommands.Tests.ScriptTests {
+    [TestClass]
+    public class RocketVolleyTests {
+        [TestMethod]
+        public void RocketVolleyTest() {
+            String script = @"
+:setup
+assign rocketGroup to 'Rockets'
+assign ""fireTickInterval"" to 1
+
+#Background Firing Loop
+async call ""fireLoop""
+call ""stopFiring""
+
+:toggle
+assign ""firing"" to not {firing}
+assign ""i"" to 0
+
+:startFiring
+assign ""firing"" to true
+assign ""i"" to 0
+
+:stopFiring
+tell [rocketGroup] rockets to not shoot
+assign ""firing"" to false
+
+:fireLoop
+if not {firing} replay
+print ""i = "" + {i}
+if {i} >= count of [rocketGroup] rockets
+  assign ""i"" to 0
+tell [rocketGroup] rockets @ {i} to shoot
+tell [rocketGroup] rockets to not shoot
+wait {fireTickInterval} ticks
+assign ""i"" to {i} + 1
+goto ""fireLoop""
+";
+            using (var test = new ScriptTest(script)) {
+                var mockRocket1 = new Mock<IMyUserControllableGun>();
+                var mockRocket2 = new Mock<IMyUserControllableGun>();
+
+                var rocket1Off = MockCalledAction(mockRocket1, "Shoot_Off");
+                var rocket1On = MockCalledAction(mockRocket1, "Shoot_On");
+                var rocket2Off = MockCalledAction(mockRocket2, "Shoot_Off");
+                var rocket2On = MockCalledAction(mockRocket2, "Shoot_On");
+
+                test.MockBlocksInGroup("Rockets", mockRocket1, mockRocket2);
+
+                test.RunOnce(); //Setup but no firing yet
+
+                rocket1Off.Verify(b => b.Apply(mockRocket1.Object));
+                rocket2Off.Verify(b => b.Apply(mockRocket2.Object));
```

Comment:
Huh! Neat. I haven't seen much with `ITerminalAction` yet, so I'm assuming that this is the only way we can actually verify that we're invoking these "functions" on these blocks. (Inconsistent space engineers APIs as opposed to the "actions" being methods on the blocks themselves?)

Reply:
Right, some functionality they do not have APIs for, there is a generic "applyAction(name)" used instead.  You'll see this in a few block handlers.